### DISCUSSION TITLE
Is this library in a mature enough state to implement into ComfyUI for handling of heif format images?

### DISCUSSION THREAD

**Discussion by shawnington** [bigcat88/pillow_heif#239] [General]
After our joint effort to solve the issue with MPO format images, i started looking at pillow support for heif formats, and found that you @bigcat88 have a library to handle that.

Your example code on pypi is: 

```
register_heif_opener()
im = Image.open("image.heic")  # do whatever need with a Pillow image
```

Does that just add heif/heic support to the Pillow functions, but still allow them to function as normally otherwise? Have any of these functions been merged into pillow? 

If I am understanding right, this opens the image as standard 8-bit RGB so it should function like a normal image when opening? 

Or will gain maps still be included?  

Im interested in a better way to filter out the gain maps from HDR image types, or if they just open as a standard 8-bit RGB image, thats also fantastic.

**Comment by bigcat88** [comment]
> Does that just add heif/heic support to the Pillow functions, but still allow them to function as normally otherwise?

Yes, that's enough.



> Have any of these functions been merged into pillow?

No, and there won't be due to licensing restrictions and a significant increase in effort on the part of the Pillow maintainers to support this.

https://github.com/python-pillow/Pillow/issues/2806

It's the same with AVIF support: bundling such a big thing into Pillow requires a lot of support work in the future.

https://github.com/python-pillow/Pillow/pull/5201



> If I am understanding right, this opens the image as standard 8-bit RGB so it should function like a normal image when opening?

Pillow supports 16-bit in "L" mode, so monochrome HEIC/AVIF files will open at 16-bit in monochrome mode.
But for ComfyUI it will be good since you do a "convert("RGB")" command for each image, which converts 16-bit monochrome color to regular 8-bit RGB.



> Or will gain maps still be included?
> 
> Im interested in a better way to filter out the gain maps from HDR image types, or if they just open as a standard 8-bit RGB image, thats also fantastic.

Bit-depth images in HEIF/AVIF files are not included in the ImageSequence and can only be obtained by calling user-defined functions.
So it's good here :)


_Is this library in a mature enough state to implement into ComfyUI for handling of heif format images?_

Yes, I think, it is already enough mature **to handle reading** of the HEIF/AVIF images, in last 4-5 months there were no reports about big problems.
